src/helpers/resolver.ts

```diff
@@ -0,0 +1,71 @@
+import { getProvider } from '@/helpers/provider';
+import { NetworkID } from '@/types';
+
+const provider = getProvider(5);
+
+type ResolvedName = {
+  networkId: NetworkID;
+  address: string;
+};
+
+function memoize<T extends any[], U>(fn: (...args: T) => U) {
+  const cache = new Map<string, any>();
+
+  return (...args: T): U => {
+    const key = JSON.stringify(args);
+
+    if (cache.has(key)) {
+      return cache.get(key);
+    }
+
+    const result = fn(...args);
+    cache.set(key, result);
+
+    return result;
+  };
+}
+
+function createResolver() {
+  const cache = new Map<string, ResolvedName | null>();
+
+  function resolveStatic(id: string): ResolvedName | null {
+    const parts = id.split(':');
+
+    return {
+      networkId: parts[0] as NetworkID,
+      address: parts[1]
+    };
+  }
+
+  async function resolveEns(id: string): Promise<ResolvedName | null> {
+    const resolvedAddress = await provider.resolveName(id);
```

Comment:
This is taking 3 requests in serie which is pretty slow, can you try instead to do a call to the function `addr()` to this contract:
```
0xd7a4F6473f32aC2Af804B3686AE8F1932bC35750
```
This is the resolver for `eth` domain on Goerli. 

You will need to pass a hash of the id as `node` like this:
```ts
import { namehash } from '@ethersproject/hash';

const name = namehash('discopolo.eth');
const address = await contract.addr(name);
```

Reply:
It's way better. Thanks.